syft/pkg/package.go

```diff
@@ -46,3 +46,21 @@ func (p Package) ID() artifact.ID {
 func (p Package) String() string {
 	return fmt.Sprintf("Pkg(name=%q version=%q type=%q id=%q)", p.Name, p.Version, p.Type, p.id)
 }
+
+func (p *Package) merge(other Package) error {
+	if p.id != other.id {
```

Comment:
Just to be sure - these are the only two conditions we would not want to merge on?

There is no use in comparing CPE either here even if they are ignored by the hash?

Reply:
> these are the only two conditions we would not want to merge on
 correct. The ID check covers all fields that would change an ID (which is most of them), which leaves CPEs and pURLs. CPEs are a slice, so they can be merged, but the pURL field is not a collection, thus cannot be merged.